platform/socket.cpp

```diff
@@ -0,0 +1,91 @@
+#include "socket.hpp"
+
+#include "base/assert.hpp"
+
+#include "std/algorithm.hpp"
+
+namespace platform {
+
+TestSocket::TestSocket()
+  : m_isConnected(false)
+{
+}
+
+TestSocket::~TestSocket()
+{
+  m_isConnected = false;
+}
+
+bool TestSocket::Open(string const & host, uint16_t port)
+{
+  m_isConnected = true;
+  return true;
+}
+
+void TestSocket::Close()
+{
+  m_isConnected = false;
+}
+
+bool TestSocket::Read(uint8_t * data, uint32_t count)
+{
+  if (!m_isConnected)
+    return false;
+
+  lock_guard<mutex> lg(m_inputMutex);
```

Comment:
If they're exists, could you please point me where they're in this CL or in our codebase?